lib/saluki-components/src/encoders/datadog/logs/mod.rs

```diff
@@ -0,0 +1,287 @@
+
+        let mut request_builder = RequestBuilder::new(
+            LogsEndpointEncoder::new(self.ddsource.clone()),
+            compression_scheme,
+            RB_BUFFER_CHUNK_SIZE,
+        )
+        .await?;
+        request_builder.with_max_inputs_per_payload(MAX_LOGS_PER_PAYLOAD);
+
+        Ok(DatadogLogs {
+            request_builder,
+            telemetry,
+        })
+    }
+}
+
+impl MemoryBounds for DatadogLogsConfiguration {
+    fn specify_bounds(&self, builder: &mut MemoryBoundsBuilder) {
+        builder.minimum().with_single_value::<DatadogLogs>("component struct");
+        builder
+            .firm()
+            .with_array::<Log>("logs split re-encode buffer", MAX_LOGS_PER_PAYLOAD);
+    }
+}
+
+pub struct DatadogLogs {
+    request_builder: RequestBuilder<LogsEndpointEncoder>,
+    telemetry: ComponentTelemetry,
+}
+
+#[async_trait]
+impl IncrementalEncoder for DatadogLogs {
+    async fn process_event(&mut self, event: Event) -> Result<ProcessResult, GenericError> {
+        let log = match event {
+            Event::Log(log) => log,
+            _ => return Ok(ProcessResult::Continue),
+        };
+
+        match self.request_builder.encode(log).await {
+            Ok(None) => Ok(ProcessResult::Continue),
+            Ok(Some(log)) => Ok(ProcessResult::FlushRequired(Event::Log(log))),
+            Err(e) => {
+                if e.is_recoverable() {
+                    warn!(error = %e, "Failed to encode Datadog log due to recoverable error. Continuing...");
+                    self.telemetry.events_dropped_encoder().increment(1);
+                    Ok(ProcessResult::Continue)
+                } else {
+                    Err(e).error_context("Failed to encode Datadog log due to unrecoverable error.")
+                }
+            }
+        }
+    }
+
+    async fn flush(&mut self, dispatcher: &PayloadsDispatcher) -> Result<(), GenericError> {
+        let maybe_requests = self.request_builder.flush().await;
+        for maybe_request in maybe_requests {
+            match maybe_request {
+                Ok((events, request)) => {
+                    let payload_meta = PayloadMetadata::from_event_count(events);
+                    let http_payload = HttpPayload::new(payload_meta, request);
+                    let payload = Payload::Http(http_payload);
+                    dispatcher.dispatch(payload).await?;
+                }
+                Err(e) => error!(error = %e, "Failed to build Datadog logs payload. Continuing..."),
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
+struct LogsEndpointEncoder {
+    ddsource: String,
+    tags_deduplicator: ReusableDeduplicator<Tag>,
+}
+
+impl LogsEndpointEncoder {
+    fn new(ddsource: String) -> Self {
+        Self {
+            ddsource,
+            tags_deduplicator: ReusableDeduplicator::new(),
+        }
+    }
+
+    fn status_to_str(status: LogStatus) -> &'static str {
```

Comment:
This would be a good method to stick on `LogStatus` itself (called `as_str`, which is the idiomatic naming for this sort of thing) that way we have it if any other code needs it.